File: backend/models/QueueEntry.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Microsoft.EntityFrameworkCore;

namespace QueueApp.Models
{
    public class QueueEntry
    {
        [Key]
        public long QueueEntryID { get; set; }  // ⚙️ Đổi lại cho đồng bộ với QueueService (entry.QueueEntryID)

        // ✅ Số lượng khách trong nhóm (1–20)
        [Required]
        [Range(1, 20, ErrorMessage = "Số người trong nhóm phải từ 1–20.")]
        public int PartySize { get; set; }

        [Required]
        public DateTime JoinTime { get; set; } = DateTime.UtcNow;

        public string? Notes { get; set; }

        // ✅ Vị trí hiện tại trong hàng chờ
        [Range(0, int.MaxValue)]
        public int? CurrentPosition { get; set; }

        // ✅ Thời gian ước tính chờ (phút)
        [Range(0, int.MaxValue)]
        [Comment("Thời gian ước tính chờ (phút)")]
        public int? EstimatedWaitTime { get; set; }

        [Required, MaxLength(50)]
        public string Status { get; set; } = "Waiting"; // "Waiting", "Called", "Canceled", "Completed"

        // 🔗 FK tới User (customer)
        [Required]
        public int UserID { get; set; }

        [ForeignKey(nameof(UserID))]
        public User User { get; set; } = null!;

        // 🔗 FK tới Restaurant
        [Required]
        public int RestaurantID { get; set; }

        [ForeignKey(nameof(RestaurantID))]
        public Restaurant Restaurant { get; set; } = null!;

        // 🔗 FK tới QueueType
        [Required]
        public int QueueTypeID { get; set; }

        [ForeignKey(nameof(QueueTypeID))]
        public QueueType QueueType { get; set; } = null!;

        // 🔗 Các thông báo liên quan (nếu có)
        public ICollection<Notification>? Notifications { get; set; }
    }
}

File: frontend/lib/models/queue_entry.dart
class QueueEntry {
  final int queueEntryID;
  final int restaurantID;
  final int userID;
  final int queueTypeID;
  final int partySize;
  final String status;
  final DateTime joinTime;
  final int? currentPosition;
  final int? estimatedWaitTime;
  final String? notes;
  final String? restaurantName;   // nếu có
  final int? queueNumber;         // nếu muốn dùng

  QueueEntry({
    required this.queueEntryID,
    required this.restaurantID,
    required this.userID,
    required this.queueTypeID,
    required this.partySize,
    required this.status,
    required this.joinTime,
    this.currentPosition,
    this.estimatedWaitTime,
    this.notes,
    this.restaurantName,
    this.queueNumber,
  });

  factory QueueEntry.fromJson(Map<String, dynamic> json) {
    return QueueEntry(
      queueEntryID: json['queueEntryID'] as int,
      restaurantID: json['restaurantID'] as int,
      userID: json['userID'] as int,
      queueTypeID: json['queueTypeID'] as int,
      partySize: json['partySize'] as int,
      status: json['status'] as String,
      joinTime: DateTime.parse(json['joinTime'] as String),
      currentPosition: json['currentPosition'] as int?,
      estimatedWaitTime: json['estimatedWaitTime'] as int?,
      notes: json['notes'] as String?,
      restaurantName: json['restaurantName'] as String?,
      queueNumber: json['queueNumber'] as int?,
    );
  }
}
